src/Webhooks/RequestBuilder.php

```diff
@@ -83,4 +86,33 @@ public function build(WebhookRequestInterface $webhookRequest): RequestInterface
             $webhookRequest->getRequestFormat()
         ));
     }
+
+    /**
+     * Validate the request object contains values that are compatible with the Diactoros request object
+     *
+     * @param \EoneoPay\Webhooks\Models\WebhookRequestInterface $webhookRequest
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Webhooks\Exceptions\InvalidRequestException
+     */
+    private function validateRequest(WebhookRequestInterface $webhookRequest): void
+    {
+        $invalidHeaders = [];
+
+        /**
+         * Whilst this check should not be required due to type-hinting, due to magic setters on the model not enforcing
+         * strict type-hinting, it is possible that this array may contain 'mixed' values
+         */
+        foreach ($webhookRequest->getRequestHeaders() as $header => $value) {
+            /** @var mixed $value */
+            if (\is_scalar($value) === false) {
+                $invalidHeaders[] = $header;
```

Comment:
Just throw in here?

Reply:
@merk throw once per invalid header structure, instead of collecting them all?

Reply:
Agreed

Reply:
You're not capturing them all, you're just throwing a generic exception with no actual detail of which headers failed.

Simpler to just throw as soon as you see a bad one?